### PR TITLE
Medibots now have the proper icons

### LINKED ISSUE #29407: Can only make white medical robots
Start with any colour medkit and it'll be the right colour all the way through assembly up until you put the last piece on, when it turns white.

I am badly triggered.

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
fix: Medibots now properly render the overlays of the medkits they are made from.
/:cl:

Because the overlays only updated on `Initialize()` something finnicky was happening that prevented the overlay from actually applying. Overlays now update in `update_icon()` and initializing adds a timer that calls `update_icon()` after a decisecond (to avoid aforementioned finnicky thing.)

Fixes #29407

### CODE REVIEW

File: code/modules/mob/living/simple_animal/bot/medbot.dm [hunk 1]
```diff
@@ -92,15 +95,11 @@
...
-	update_icon()
-
-	if(skin)
-		add_overlay("medskin_[skin]")
-
 	var/datum/job/doctor/J = new /datum/job/doctor
 	access_card.access += J.get_access()
 	prev_access = access_card.access
 	qdel(J)
+	addtimer(CALLBACK(src, .proc/update_icon), 1)
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/bot/medbot.dm]: there should be no need to delay this.

Comment (Xhuis) [inline code/modules/mob/living/simple_animal/bot/medbot.dm]: I assumed as much, but strangely it didn't work unless I did.

**Comment by FantasticFwoosh** [comment]
I wasn't even aware they overlayed, i thought they were all 1 sprite.

File: code/modules/mob/living/simple_animal/bot/medbot.dm (hunk 1 above)
Comment (lzimann) [inline code/modules/mob/living/simple_animal/bot/medbot.dm]: This addtimer makes absolutely no sense.

Comment (Xhuis) [inline code/modules/mob/living/simple_animal/bot/medbot.dm]: Neither does the fact that it doesn't work without it, but it is what it is.

Comment (tortellinitony) [inline code/modules/mob/living/simple_animal/bot/medbot.dm]: Late initialize?

Late initialize machine broke

Comment (Xhuis) [inline code/modules/mob/living/simple_animal/bot/medbot.dm]: Understandable have a great day

Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/bot/medbot.dm]: well clearly the issue is that it's setting the skin after Initialize() is called, so it never shows up.
Accordingly, make the skin an argument for Initialize().

Comment (Xhuis) [inline code/modules/mob/living/simple_animal/bot/medbot.dm]: Hmm, that should fix it. I'm on mobile but I'll do that once I return to Casa de Xhuis.

**Pull request merged**